python/sglang/srt/model_loader/loader.py

```diff
@@ -145,11 +145,20 @@ def _initialize_model(
     load_config: LoadConfig,
 ) -> nn.Module:
     """Initialize a model with the given configurations."""
-    model_class, _ = get_model_architecture(model_config)
+    model_class, model_arch = get_model_architecture(model_config)
     packed_modules_mapping = getattr(model_class, "packed_modules_mapping", {})
     quant_config = _get_quantization_config(
         model_config, load_config, packed_modules_mapping
     )
+    if (
+        quant_config is not None
+        and quant_config.get_name() == "modelopt_fp4"
+        and model_arch == "DeepseekV3ForCausalLMNextN"
+    ):
+        logger.warning(
+            "Overriding DeepseekV3ForCausalLMNextN quant config for modelopt_fp4 Deepseek model."
+        )
+        quant_config = None
```

Comment:
It is better to move it into `DeepseekV2ForCausalLM::__init__` after applying this https://github.com/sgl-project/sglang/pull/7327/files#r2159861879